currency_monthly_rate/tests/test_currency_monthly_rate.py

```diff
@@ -0,0 +1,72 @@
+# Copyright 2018 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo.tests.common import TransactionCase
+from odoo import fields
+
+
+class TestCurrencyMonthlyRate(TransactionCase):
+
+    def setUp(self):
+        super(TestCurrencyMonthlyRate, self).setUp()
+
+        self.usd = self.env.ref('base.USD')
+        self.eur = self.env.ref('base.EUR')
+
+        self.year = str(fields.Date.from_string(fields.Date.today()).year)
+
+        monthly_rate = self.env['res.currency.rate.monthly']
+        rate = self.env['res.currency.rate']
+
+        monthly_rates_to_create = [
+            {'month': '01', 'rate': 1.20},
+            {'month': '02', 'rate': 1.40},
+        ]
+        for r in monthly_rates_to_create:
+            r.update({
+                'year': self.year,
+                'currency_id': self.usd.id
+            })
+            monthly_rate.create(r)
+
+        rates_to_create = [
+            {'name': '%s-01-01' % self.year, 'rate': 1.15},
+            {'name': '%s-01-11' % self.year, 'rate': 1.10},
+            {'name': '%s-01-31' % self.year, 'rate': 1.35},
+            {'name': '%s-02-01' % self.year, 'rate': 1.50},
+            {'name': '%s-02-11' % self.year, 'rate': 1.30},
+        ]
+        for r in rates_to_create:
+            r.update({
```

Comment:
same, `self.update` is `@api.multi`

Reply:
the update is on dictionnary and not on recordset :confused:

Reply:
Whoa, that's right
I must have been asleep at while writing that comment :stuck_out_tongue_winking_eye: